google/auth/_agent_identity_utils.py

```diff
@@ -0,0 +1,165 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Helpers for Agent Identity credentials."""
+
+import base64
+import hashlib
+import os
+import re
+import time
+
+from google.auth import environment_vars
+from google.auth import exceptions
+
+# SPIFFE trust domain patterns for Agent Identities.
+_AGENT_IDENTITY_SPIFFE_TRUST_DOMAIN_PATTERNS = [
+    r"^agents\.global\.org-\d+\.system\.id\.goog$",
+    r"^agents\.global\.proj-\d+\.system\.id\.goog$",
+]
+
+
+def get_agent_identity_certificate_path():
+    """Gets the certificate path from the certificate config file.
+
+    The path to the certificate config file is read from the
+    GOOGLE_API_CERTIFICATE_CONFIG environment variable. This function
+    implements a retry mechanism to handle cases where the environment
+    variable is set before the file is available on the filesystem.
+
+    Returns:
+        str: The path to the leaf certificate file.
+
+    Raises:
+        google.auth.exceptions.RefreshError: If the certificate config file
+            or the certificate file cannot be found after retries.
+    """
+    import json
+
+    cert_config_path = os.environ.get(environment_vars.GOOGLE_API_CERTIFICATE_CONFIG)
+    if not cert_config_path:
+        return None
+
+    # Poll for the config file and the certificate file to be available.
+    # Phase 1: Poll rapidly for 5 seconds (50 * 0.1s).
+    # Phase 2: Slow down polling for the next 25 seconds (50 * 0.5s).
+    for i in range(100):
+        if os.path.exists(cert_config_path):
+            with open(cert_config_path, "r") as f:
+                cert_config = json.load(f)
+                cert_path = (
+                    cert_config.get("cert_configs", {})
+                    .get("workload", {})
+                    .get("cert_path")
+                )
+                if cert_path and os.path.exists(cert_path):
+                    return cert_path
+        if i < 50:
+            time.sleep(0.1)
+        else:
+            time.sleep(0.5)
```

Comment:
nit: maybe instead of using conditional sleeps here, you can generate the sleep values directly? Something like:

BACKOFF_VALUES = [0.1] * 50 + [0.5] * 50

for sleep_value in BACKOFF_VALUES:
   ...
   time.sleep(sleep_value)